packages/core/src/lib/tooltip/tooltip-styles.ts

```diff
@@ -73,12 +76,28 @@ const createContext = (): TooltipContext => {
   );
   const style = floatingStyle({ offset: 7, shift: 5, flip: true, auto: true });
 
+  const { set: setHoveredTimeout, clear: clearHoveredTimeout } = useTimeout();
```

Comment:
When I was building this out, I had imagined any delays happening as part of the `isVisible` derived store. `isHovered`, IMO, should track the actual "is this item hovered?" state, while a delay should be incorporated into the logic of "knowing this is hovered, should it be visible?".

If `hoverDelayMS` becomes a store, this also gives us slightly better reactivity / cleanup if the delay value changes, which may be a thing we want to do in the future (e.g. for a group of icons, once the tooltip displays for one of them, we should not use a delay for other icons in the group).

```ts
const hoverDelayMS = writable(0);  
const isVisible = derived(
  [isHovered, visibility, hoverDelayMS],
  ([$isHovered, $visibility, $hoverDelayMS], set) => {
    let cleanup = noop;

    if ($visibility === 'visible') {
      set(true);
    } else if ($visibility === 'invisible' || !$isHovered) {
      set(false);
    } else {
      const timeoutID = setTimeout(() => set(true), $hoverDelayMS);
      cleanup = () => clearTimeout(timeoutID);
    }

    return cleanup;
  },
  false
);
```

Reply:
Thanks for the suggestion on this - much cleaner and better reactivity.